doc/mgmt/gnmi/gnoi_os_reboot_file_reset.md

```diff
@@ -0,0 +1,449 @@
+# gNOI HLD for OS, Reboot, File, Factory Reset APIs #
+
+# High Level Design Document
+
+#### Rev 0.1
+
+## Table of Contents
+- [Table of Contents](#table-of-contents)
+- [Revision](#revision)
+- [Scope](#scope)
+- [Definition/Abbreviation](#definitionabbreviation)
+- [Overview](#overview)
+- [Requirements](#2-requirements)
+- [Architecture Design](#3-architecture-design)
+- [High Level Design](#4-high-level-design)
+- [SAI API](#sai-api)
+- [Warmboot and Fastboot Design Impact](#warmboot-and-fastboot-design-impact)
+- [Restrictions/Limitations](#restrictionslimitations)
+- [Testing Requirements/Design](#testing-requirementsdesign)
+- [Open/Action items - if any](#openaction-items---if-any)
+
+### Revision
+
+| Rev  | Rev Date   | Author(s)          | Change Description |
+|------|------------|--------------------|--------------------|
+| v0.1 | 07/22/2024 | Neha Das (Google)   | Initial version |
+
+## Scope
+
+This document describes the high level design of adding the capability to support gNOI APIs for OS, Reboot, File, Factory Reset services in the telemetry server. 
```

Comment:
Can we be a bit more specific about what RPC exactly (maybe make a list) that we implement? For example, OS has three RPCs but Reboot is a RPC under System service, so it doesn't make that much sense to put them side by side.